src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Builder/OpenTelemetryBuilderOtlpExporterExtensions.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Exporter;
+
+/// <summary>
+/// Contains extension methods to facilitate registration of the OpenTelemetry
+/// Protocol (OTLP) exporter into an <see cref="IOpenTelemetryBuilder"/>
+/// instance.
+/// </summary>
+public static class OpenTelemetryBuilderOtlpExporterExtensions
+{
+    private const int DefaultProcessorPipelineWeight = 10_000;
+
+    /// <summary>
+    /// Uses OpenTelemetry Protocol (OTLP) exporter for all signals.
+    /// </summary>
+    /// <remarks>
+    /// Notes:
+    /// <list type="bullet">
+    /// <item>Calling this method automatically enables logging, metrics, and
+    /// tracing.</item>
+    /// <item>The exporter registered by this method will be added as the last
+    /// processor in the pipeline established for logging and tracing.</item>
+    /// <item>This method can only be called once. Subsequent calls will results
+    /// in a <see cref="NotSupportedException"/> being thrown.</item>
+    /// <item>This method cannot be called in addition to signal-specific
+    /// <c>AddOtlpExporter</c> methods. If this method is called signal-specific
+    /// <c>AddOtlpExporter</c> calls will result in a <see
+    /// cref="NotSupportedException"/> being thrown.</item>
+    /// </list>
+    /// </remarks>
+    /// <param name="builder"><see cref="IOpenTelemetryBuilder"/>.</param>
+    /// <returns>Supplied <see cref="IOpenTelemetryBuilder"/> for chaining calls.</returns>
+    public static IOpenTelemetryBuilder UseOtlpExporter(
+        this IOpenTelemetryBuilder builder)
+        => UseOtlpExporter(builder, name: null);
+
+    /// <summary><inheritdoc cref="UseOtlpExporter(IOpenTelemetryBuilder)"/></summary>
+    /// <remarks><inheritdoc cref="UseOtlpExporter(IOpenTelemetryBuilder)" path="/remarks"/></remarks>
+    /// <returns><inheritdoc cref="UseOtlpExporter(IOpenTelemetryBuilder)" path="/returns"/></returns>
+    /// <param name="builder"><see cref="IOpenTelemetryBuilder"/>.</param>
+    /// <param name="baseEndpoint">The base endpoint to use. A signal-specific
+    /// path will be appended to the base endpoint for each signal
+    /// automatically if the protocol is set to <see cref="OtlpExportProtocol.HttpProtobuf"/>.</param>
+    public static IOpenTelemetryBuilder UseOtlpExporter(
```

Comment:
This looks like grpc protocol specific method to me. Do we need this as a separate method?

Reply:
I'm not following. What do you mean by grpc-specific? At the moment there are 3 overloads exposed:

```csharp
// Everything is default or comes from IConfiguration/envvars
appBuilder.Services.AddOpenTelemetry()
    .UseOtlpExporter();

// Specify the endpoint. Everything else is default or comes from IConfiguration/envvars
appBuilder.Services.AddOpenTelemetry()
    .UseOtlpExporter(new Uri("http://my_otlp_endpoint/"));

// Specify the protocol + endpoint. Everything else is default or comes from IConfiguration/envvars
appBuilder.Services.AddOpenTelemetry()
    .UseOtlpExporter(OtlpExportProtocol.HttpProtobuf, new Uri("http://my_otlp_endpoint/"));
```

Reply:
for this particular case

```
appBuilder.Services.AddOpenTelemetry()
    .UseOtlpExporter(new Uri("http://my_otlp_endpoint/"));
```

Are expecting many users specifying the endpoint here and changing the protocol via environment variable?

Reply:
@vishweshbankwar 

I chatted with @alanwest a bit. We decided to tweak the overloads. They are now:

```
UseOtlpExporter();
UseOtlpExporter(OtlpExportProtocol protocol);
UseOtlpExporter(OtlpExportProtocol protocol, Uri baseEndpoint);
```

How you feel about that? TBH we weren't sure exactly what concern you were raising but these seemed more useful in the end 😄

Reply:
Chatted offline with @vishweshbankwar. We decided to drop the second overload. What we have now is:

```
UseOtlpExporter();
UseOtlpExporter(OtlpExportProtocol protocol, Uri baseEndpoint);
```